indico/modules/events/registration/models/registrations.py

```diff
@@ -822,6 +910,29 @@ def generate_ticket_apple_wallet(self):
         if awm.is_configured:
             return awm.get_ticket(self)
 
+    def can_perform_check(self, *, check_type=None, is_check_out=False):
+        can_check = True
+        reason = _('Check can be performed.')
+        if not check_type:
+            check_type = self.event.default_check_type if self.event else None
+        if not check_type:
+            can_check = False
+            reason = _('Check type not specified and no default check type has been set for this event.')
+            return {'can_check': can_check, 'reason': reason}
+        if is_check_out and not check_type.check_out_allowed:
+            can_check = False
+            reason = _('This check type cannot be "checked-out".')
+        previous_checks_for_type = [check for check in self.checks if check.check_type == check_type and
+                                    check.is_check_out == is_check_out]
+        if previous_checks_for_type and check_type.rule == RegistrationCheckRule.once:
+            can_check = False
+            reason = _('This check type can only be performed once.')
+        if (check_type.rule == RegistrationCheckRule.once_daily and
+                any(check.timestamp.date() == now_utc().date() for check in previous_checks_for_type)):
```

Comment:
you need to convert to the event's tzinfo; otherwise you always do this comparison using the date in UTC, instead of the correct date in the event's timezone